hack/local-development/start-extension-provider-local

```diff
@@ -53,7 +53,13 @@ if [ "$USER" != root ]; then
   SUDO="sudo -E"
 fi
 
-$SUDO go run \
+namespace="$(kubectl --kubeconfig "$kubeconfig" get ns -l controllerregistration.core.gardener.cloud/name=provider-local -o jsonpath={..metadata.name})"
+if [[ -z "$namespace" ]]; then
+  echo "gardenlet has not yet created the namespace for the extension, cannot start provider-local"
+  exit 1
+fi
```

Comment:
We also had in mind similar change in https://github.com/gardener/gardener/issues/7237. But as explained in https://github.com/gardener/gardener/issues/7237#issuecomment-1440248730, there is a provider-local specific logic that expects the namespace to be empty string.

https://github.com/gardener/gardener/blob/afe07938c905c47cef81565ebafc1c203dc96978/extensions/pkg/webhook/cmd/options.go#L277-L282

We didn't know the context about this handling.

Reply:
This is not provider-local specific but part of the extensions library and applicable for every extension not setting the namespace. Anyways, the motivation here was that the secrets manager cannot be used without a namespace where it could store the generated secrets. Hence, when the namespace is not provided, the secrets manager is not used and instead the certificates are generated directly and written to a disk.

Reply:
Ok. So it should work when provider-local is running locally and this certificate management reconciler is running (added to the manager) (instead of unmanaged certificates generated locally)?

Reply:
With this PR, we pass the namespace for the webhook, so the certificates can be managed by secrets manager and stored in this namespace. TBH, I think this could have built this way from day one. I think we missed it back then. Meanwhile, the webhook server namespace should be provided, and even earlier it was necessary to cleanup the webhook configuration in the seed cluster after deletion. Perhaps we can even drop this special handling code you linked above (though, this requires a dedicated evaluation, I believe).